### DISCUSSION TITLE
Input arguments and close mosaic option

### DISCUSSION THREAD

**Discussion by Marco-Nguyen** [ultralytics/ultralytics#820] [Q&A] [answered]
Hi, I would like to ask 2 questions:

1. I am using Python. Where can I find the list of full input arguments for `model.train()`, `model.val()`, `model.predict()`, and `model.export()`? 
&nbsp;&nbsp;&nbsp;I have searched in the repo, official docs website but the provided lists do not contain all arguments. For example, device argument can be used in prediction or validation but not listed.

3. What is the purpose of disabling mosaic augmentation for some final epochs? Is there any guidance on how to set the value?  and where can I read more on this?

**Comment by Laughing-q** [comment, marked as answer]
@Marco-Nguyen hi you can find the args here. [https://github.com/ultralytics/ultralytics/blob/main/ultralytics/yolo/cfg/default.yaml](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/yolo/cfg/default.yaml) or in our docs [https://docs.ultralytics.com/cfg/](https://docs.ultralytics.com/cfg/).
About the `close-mosaic`, the idea is from YOLOX and we found it's also benefiting our model. We set `close-mosaic=10` as default, I think mostly you don't need to modify it, but if you're using less epochs like 50 epochs or less, then I recommend you modify it to a small number .i.e 5.

**Reply by Marco-Nguyen** [reply]
@Laughing-q Thanks for the response.

For question 1, as I mentioned, `device` is not listed neither in the docs nor `default.yaml` of predict and val. Also in predict, I can pass `conf` to change the default confidence threshold, and it is not listed as well. Will you update this in docs?

**Reply by Laughing-q** [reply]
@Marco-Nguyen sure! :)
